### ISSUE TITLE
Settings dialog, TTS voices page: select default for language + gender

### ISSUE THREAD

**Issue by marisademeglio** [daisy/pipeline-ui#236]
For a next release: it might be useful to be able to select one default voice for each language/gender combination.

_Originally posted by @bertfrees in https://github.com/daisy/pipeline-ui/issues/170#issuecomment-2159257710_

**Comment by marisademeglio** [comment]
@bertfrees So within a language: one for male, one for female, one for *? Or if there is one for * then that overrides the ability to select one for male/female? 

And this is at the language/gender level, not considering dialect, correct?

**Comment by bertfrees** [comment]
For me it depends on the user interface. This seems like a useful feature, if we can find a way to do it without compromising on simplicity. Considering dialect would be a good addition too, I think.

Have you got an idea for how to do it in the user interface?

What exactly do you mean with "*"? Do you mean specifying a voice as the default for a language, no matter which gender is requested? I'm not sure that would be useful.

**Comment by marisademeglio** [comment]
I was thinking of how the voices are categorized: male, female, or unknown (which I think used to be labelled "*").  But looking at it from the production side, how is gender indicated?

I think the best way for the UI is to add a new settings section (in the navigation to the left, under "Preferred voices") per language and offer more thorough management of defaults wrt gender and dialect. I've thought through some different approaches, and I really believe that making small changes to the current interface, which purposefully left out dialect (it was an explicit request) can quickly result in a confusing set of controls. I'm also thinking this feature should probably be pushed to the next release. 

For reference, this is the current preferred voices dialog.

<img width="1112" height="912" alt="preferred voices settings dialog" src="https://github.com/user-attachments/assets/6a77f055-fbf0-44d0-8062-3f81269cc60c" />

**Comment by bertfrees** [comment]
Some voices are internally categorized as "*" indeed. It can mean two things: either the provider of the voices marks them as "gender neutral", or because we simply don't know the gender.

Pipeline tries to match available voices with what is needed by the document, which can also have a style sheet attached (and taking into account the user's preferences). The requested gender and age are specified in CSS: https://www.w3.org/TR/css-speech-1/#typedef-voice-family-gender (`male`/`female`/`neutral`) and https://www.w3.org/TR/css-speech-1/#typedef-voice-family-age.

So in theory, we could indeed have a default for `neutral`. I'm just not sure how useful it would be in practice. What exactly makes a voice neutral? Also, would we suggest making a voice the default for a certain gender based on the internal categorization, or can the user freely choose the gender? The former would be in line with how it works for language. But the latter also makes sense, e.g. in order to allow using a "young boy" voice as a "female adult" or visa versa. Also a problem with the former is that voices categorized as "*" may not actually be _neutral_, but _unknown_ (could be male or female). Perhaps we can fix this though.

I agree that this needs more thought, so pushing to the next release is a good idea. However I think we need to be careful with adding more settings sections for voices. We already have two.

To shine a different light on this issue, perhaps we should first take a step back and think about what it means to be "the default voice for a language + gender(/age)", and which other defaults we might need. The "The default voice for English is ..." text is a simplification/approximation. It basically just means that this voice will be selected when no other voice is a better match, regardless of whether region/dialect is taken into account (and bearing in mind that voice name, engine, gender and age are more important than region and priority). Behind the scenes, we are making the region a wildcard, and we increase the priority. Following this pattern, allowing the user to set more than one default for each language (one for each gender), basically just means allowing the user to prioritize more voices.

What I'm trying to say is this: what if, instead of providing more settings in new settings sections, we stick with [the original concept](https://github.com/daisy/pipeline-ui/issues/169) of a single "prioritize" action per voice, and we change how the radio buttons work, and the way we explain to the user what his actions mean.

I'd like to come back to an idea I expressed [before](https://github.com/daisy/pipeline-ui/issues/170#issuecomment-2147207769):

> The list of preferred voices basically represents the voice config XML. In that regard, I thought it might be a good idea to present it in such a way that it makes it more obvious how the voice selection will work. This is currently left up to the user to guess, and I think it's pretty intuitive so should be fine, but still... The user might not realize at first that an American English preferred voice will be used for any English content (unless another preferred voice is selected as default). So perhaps each preferred voice could have a column or ℹ button that tells you in plain English when the voice will be used, similar to how I did it [in my comment above](https://github.com/daisy/pipeline-ui/issues/170#issuecomment-2142301299).

In other words: we replace the "The default voice for English is ..." text at the bottom of the language section, with one "This voice is used when ..." text for each voice. When you click a radio button, it would not automatically deselect _all_ other buttons in the group, but it would deselect only what needs to be deselected to solve conflicts. (E.g. you can not prioritize two voices with the same language _and_ the same gender.) The user could get warnings about voices that are automatically deselected.

**Comment by ways2read** [comment]
I have also seen "nonbinary" and "child" used. I have adopted the label "persona" rather than "gender" in the [TTS explorer](https://dl.daisy.org/projects/tts-demo/tts_evaluation.html).

**Comment by marisademeglio** [comment]
Ok sounds like there are some good things to discuss, let's pick this back up during the next development cycle.

**Comment by marisademeglio** [comment]
> However I think we need to be careful with adding more settings sections for voices. We already have two.

@bertfrees What if we had an entirely separate dialog just for TTS settings? Just an idea. It could have these sections

* External TTS Services
* Browse Voices
* Preferred Voices
    * English
    * Spanish
    * Russian 
    * ...
* TTS Options

where each "Preferred" language can have more room to describe to the user the impact of their choices, and more room for using controls that make sense (also from an accessibility standpoint, we shouldn't change how they behave). 

And then that frees up some space to make the main settings dialog simpler, e.g. I can move some stuff out of "General" (it's getting too crowded) and add a section about the external pipeline engine for #315.

**Comment by bertfrees** [comment]
I'm OK with an entirely separate dialog just for TTS settings. Might indeed be cleaner.

Whether we need one section per language, I'm not sure about.

> also from an accessibility standpoint, we shouldn't change how they behave

You mean not change how e.g. a radio button works, or not change the current way of selecting default voices?
